demo-app/src/index.js

```diff
@@ -52,6 +52,9 @@ router.registerClients(
     return clientMap;
   }, {})
 );
+router.setEnvData({
```

Comment:
The other option is to have this as a part of the registerClients method.  I wasn't sure if anyone had a preference.  Or we could also add attributes to the custom element, but i know working with objects in attributes is a little odd.

Reply:
I think making it part of `registerClients` is a good idea, although that might mean re-naming that function to something else. That would let us guarantee that the environment data is in place and ready to go before any clients even start to load. Without that guarantee I can see scenarios where a host and a client could get into weird race conditions where sometimes the environment data would be available and sometimes not.